crates/ruff/src/rules/pycodestyle/rules/trailing_whitespace.rs

```diff
@@ -85,10 +89,30 @@ pub(crate) fn trailing_whitespace(line: &Line, settings: &Settings) -> Option<Di
         if range == line.range() {
             if settings.rules.enabled(Rule::BlankLineWithWhitespace) {
                 let mut diagnostic = Diagnostic::new(BlankLineWithWhitespace, range);
+
                 if settings.rules.should_fix(Rule::BlankLineWithWhitespace) {
-                    #[allow(deprecated)]
-                    diagnostic.set_fix(Fix::unspecified(Edit::range_deletion(range)));
+                    // If this line is blank with whitespace, we have to ensure that the previous line
+                    // doesn't end with a backslash. If it did, the file would end with a backslash
+                    // and therefore have an "unexpected EOF" SyntaxError, so we have to remove it.
+                    if let Some(prev) = prev_line {
+                        let trimmed = prev.trim_end();
+                        if trimmed.ends_with('\\') {
+                            let initial_len = prev.text_len();
+                            diagnostic.range = range.sub_start(
+                                // Shift by the amount of whitespace in the previous line, plus the
+                                // newline, plus the slash, plus any remaining whitespace after it.
```

Comment:
Is the `TextSize::from(1)` also correct in the case of `\r\n`? Similarly, could that computation be simplified by using the start of `prev` plus the trimmed length as start of the range?

Reply:
I've simplified the calculation. Another question is - do we want the space between `; \` removed as well? Technically, that should get removed by `Rule::TrailingWhitespace`, not `Rule::BlankLineWithWhitespace`. Curious what you think @konstin cc @charliermarsh